features/profile/src/main/kotlin/com/teacherworkout/features/profile/settings/UpdateProfilePictureDialog.kt

```diff
@@ -0,0 +1,97 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.teacherworkout.features.profile.R
+
+@Composable
+fun UpdateProfilePictureDialog(
+    isOpen: Boolean,
+    onCloseRequest: () -> Unit,
+    onDismissRequest: () -> Unit,
+    onNewPicture: (Uri) -> Unit
+) {
+    val screenWidth = LocalConfiguration.current.screenWidthDp.dp
+
+    val imagePicker = rememberImagePicker {
+        onNewPicture(it)
+    }
+
+    if(isOpen) {
+        Dialog(onDismissRequest = onDismissRequest) {
+            val space24dp = dimensionResource(id = R.dimen.space_24dp)
+            val corner8dp = dimensionResource(id = R.dimen.corner_8dp)
+            val minTouchSize = dimensionResource(id = R.dimen.min_touch_size)
+
+            Surface(shape = RoundedCornerShape(corner8dp)) {
+                Column(
+                    modifier = Modifier.padding(space24dp),
+                    verticalArrangement = Arrangement.spacedBy(space24dp),
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    Box(
+                        modifier = Modifier
+                            .size(minTouchSize)
+                            .clip(CircleShape)
+                            .background(MaterialTheme.colors.primary),
+                        contentAlignment = Alignment.Center
+                    ) {
+                        Icon(
+                            imageVector = Icons.Outlined.AddPhotoAlternate,
+                            tint = MaterialTheme.colors.onPrimary,
+                            //TODO: update content description of the 'add photo' icon
+                            contentDescription = null
+                        )
+                    }
+                    Text(
+                        modifier = Modifier.widthIn(max = screenWidth / 2),
+                        text = stringResource(R.string.label_change_picture),
+                        textAlign = TextAlign.Center,
+                        style = TextStyle(
+                            fontSize = MaterialTheme.typography.h6.fontSize,
+                            fontWeight = FontWeight.Bold
+                        )
+                    )
+                    Button(
+                        modifier = Modifier
+                            .height(minTouchSize)
+                            .fillMaxWidth(),
+                        onClick = {
+                            onCloseRequest()
+                            imagePicker.launch()
+                        },
+                        shape = RoundedCornerShape(50)
+                    ) {
+                        Text(text = stringResource(R.string.label_upload_picture))
+                    }
+                    //TODO: maybe this should be converted to an actual Button, because currently the click region is very small
```

Comment:
Yeah, that's a good idea! It's pretty hard now to click on the cancel button. Let's do that.